src/transpose.rs

```diff
@@ -12,9 +12,7 @@ pub(crate) fn transpose(output: &mut [u64; 64], input: [u64; 64]) {
         [0x00000000FFFFFFFF, 0xFFFFFFFF00000000],
     ];
 
-    for h in 0..64 {
-        output[h] = input[h];
-    }
+    output[..64].copy_from_slice(&input[..64]);
```

Comment:
This can probably be simplified to just `*output = input`.

`input` and `output` have the exact same type. This conversion to slices will incur bounds checking that might or might not be optimized away. Either way, it feels like lots of duplication to specify `64` in two more places than needed.

Reply:
yes this can be simplified, agreed :) !